src/transformation/visitors/function.ts

```diff
@@ -53,8 +52,10 @@ function isRestParameterReferenced(identifier: lua.Identifier, scope: Scope): bo
 
 export function transformFunctionBodyContent(context: TransformationContext, body: ts.ConciseBody): lua.Statement[] {
     if (!ts.isBlock(body)) {
+        context.pushPrecedingStatements();
         const returnStatement = transformExpressionBodyToReturnStatement(context, body);
-        return [returnStatement];
+        const precedingStatements = context.popPrecedingStatements();
+        return [...precedingStatements, returnStatement];
```

Comment:
I see this same pattern recurring a lot, it might be nice to refactor this to something like this?
```ts
function transformInPrecedingStatementScope(context: TransformationContext, transformer: () => lua.Statement | lua.Statements[]) {
        context.pushPrecedingStatements();
        const statementOrStatements = transformer();
        const precedingStatements = context.popPrecedingStatements();
        return [precedingStatements, statementOrStatements];
}
```
So this function could write just
```ts
const [precedingStatements, returnStatement] = transformInPrecedingStatementScope(context, () => transformExpressionBodyToReturnStatement(context, body));
```
(Assuming you could reuse the same refactor at other occurrences of this push/pop/concat pattern)

Reply:
Good call. I was able to replace a ton of instances of this pattern with the helper.